### PR TITLE
Msbuild fixes

### LINKED ISSUE #642: Restructure MSBuild script
The NUnit.proj script is getting quite cumbersome. Because everything is in a global namespace, it's quite easy to get conflicts when there are multiple updates to the file. We should look at some ways to structure this into mutliple files.

### LINKED ISSUE #649: Change how we zip packages
Our msbuild script uses Git Export to create zip packages. This is a bit fragile and in the case of binaries actually modifies the image directories.

We should either use a community Task to zip or create an inline task if we don't want to add any more pre-requisites for building NUnit.

### PR DESCRIPTION
**Pull request by CharliePoole**
This pull request fixes two issues with the build and partially fixes one other:

Fixes #642 It turns out not to be possible to split the script into multiple imported files due to a bug in Mono xbuild. Instead, I have added Region directives, which make it a bit easier to find specific targets provided the necessary extension is installed in the IDE.  Since this seems to be as far as we can go on the issue, we can close it once we merge this.

Fixes #649 We now use a community task to zip binaries, rather than git. We still use Git to zip up the source.

#760 Source and zip files are built on Linux now. NuGet packages are skipped due to an error that was arising. Work will continue on this issue in a separate branch after we make some decisions about what packages should be built.

### CODE REVIEW

**Comment by CharliePoole** [comment]
@chris-smith-zocdoc You may want to build on this once it's merged for the NuGet restore. For one thing, it updates the version of nuget we are running.

**Comment by chris-smith-zocdoc** [comment]
@CharliePoole Ah yes, I had done the same thing in my branch. I'll rebase on top of this once it is merged

**Comment by CharliePoole** [comment]
@nunit/core-team Would somebody please review this? I know it hasn't been here that long but it's blocking other work. I could just merge it but I feel like I should follow the rules. @rprouse has been a real champ in doing most of the reviews of my stuff in the past but he is tied up with a new job and it would be great for somebody else to step up to the plate! Remember, we have this rule that it takes two committers to approve a commit to master!

File: NUnit.proj [hunk 1]
```diff
@@ -407,22 +414,12 @@
     <Message Text="* Creating the binary $(Configuration) package as $(PackageName)$(ConfigSuffix).zip" />
     <Message Text="******************************************************************" />
 
-    <CallTarget Targets="_RemoveGitDirFromImage" />
```
Comment (chris-smith-zocdoc) [inline NUnit.proj]: Can the target `_RemoveGitDirFromImage` be removed? I don't see any other references to it.

**Comment by chris-smith-zocdoc** [comment]
> provided the necessary extension is installed in the IDE

What is the name of the extension? How do we plan on authors discovering such an extension can be used?

File: NUnit.proj [hunk 2]
```diff
@@ -501,18 +498,9 @@
...
-
-    <Exec Command='git init "$(CurrentImageDir)"' />
-    <Exec WorkingDirectory='$(CurrentImageDir)' Command='git add *.txt nunit.ico' />
-    <Exec WorkingDirectory='$(CurrentImageDir)' Command='git add bin\netcf-3.5\*' />
-    <Exec WorkingDirectory='$(CurrentImageDir)' Command='git config user.name "temp"' />
-    <Exec WorkingDirectory='$(CurrentImageDir)' Command='git config user.email "temp@temp.com"' />
-    <Exec WorkingDirectory='$(CurrentImageDir)' Command='git commit -m "Export"' />
-    <Exec WorkingDirectory='$(CurrentImageDir)'
-      Command='git archive -o "$(ProjectPackageDir)\$(PackageNameCF)$(ConfigSuffix).zip" HEAD'></Exec>
+    <Zip Files="@(CFBinZipFiles)"
```
Comment (chris-smith-zocdoc) [inline NUnit.proj]: Above in the `PackageBinaries` target you set ` WorkingDirectory="$(CurrentImageDir)"` is that not necessary here?

File: packages/repositories.config [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <repositories>
+  <repository path="../src/NUnitEngine/Addins/nunit.v2.driver.tests/packages.config" />
   <repository path="../src/NUnitFramework/testdata/packages.config" />
   <repository path="../src/NUnitFramework/tests/packages.config" />
   <repository path="..\src\NUnitEngine\Addins\nunit.v2.driver.tests\packages.config" />
```
Comment (chris-smith-zocdoc) [inline packages/repositories.config]: This file has mixed slashes, should you standardize these as well?

File: NUnit.proj (hunk 1 above)
Comment (CharliePoole) [inline NUnit.proj]: Right, it can.

File: NUnit.proj (hunk 2 above)
Comment (CharliePoole) [inline NUnit.proj]: Good call.

File: packages/repositories.config (hunk 3 above)
Comment (CharliePoole) [inline packages/repositories.config]: Seems never to have been a problem. However, I'm not sure why the repositories.config file is changing at all as a part of this, so I'll check. Also, the added line is actually the same path as the line three below it!

**Comment by CharliePoole** [comment]
@chris-smith-zocdoc I'll fix and merge.

**Comment by CharliePoole** [comment]
The extension that allows collapsing regions in XML is the XAML Region extension on the VS gallery. It's referenced in issue #642.

**Comment by CharliePoole** [comment]
Regarding repositories.config. I think we may want to fix this using a separate branch and PR, just to be sure. We can do that either before or after you finish the work on package restore.

**Pull request merged**